addons/stock/models/stock_move.py

```diff
@@ -2017,10 +2017,10 @@ def _skip_push(self):
         )
 
     def _check_quantity(self):
-        return self.env['stock.quant'].search([
+        return self.env['stock.quant'].sudo().search([
             ('product_id', 'in', self.product_id.ids),
             ('location_id', 'child_of', self.location_dest_id.ids),
-            ('lot_id', 'in', self.lot_ids.ids)
+            ('lot_id', 'in', self.sudo().lot_ids.ids)
```

Comment:
These `sudo`s can introduce cross company search result/calculation errors

Reply:
As I read it `check_quantity` checks that serial numbers are not present multiple times in locations. Which is something that can be done in sudo. Restrictions on companies do not matter in that case.